azuredevops/internal/service/policy/common.go

```diff
@@ -63,7 +64,63 @@ func genBasePolicyResource(crudArgs *policyCrudArgs) *schema.Resource {
 		Update:   genPolicyUpdateFunc(crudArgs),
 		Delete:   genPolicyDeleteFunc(crudArgs),
 		Importer: tfhelper.ImportProjectQualifiedResourceInteger(),
-		Schema:   genBaseSchema(),
+		Schema: map[string]*schema.Schema{
+			SchemaProjectID: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+			SchemaEnabled: {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			SchemaBlocking: {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			SchemaSettings: {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						SchemaScope: {
+							Type: schema.TypeList,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									SchemaRepositoryID: {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validate.NoEmptyStrings,
+									},
+									SchemaRepositoryRef: {
+										Type:         schema.TypeString,
+										Optional:     true,
+										RequiredWith: []string{SchemaRepositoryID},
```

Comment:
```suggestion
										RequiredWith: []string{"settings.0.scope.0.repository_id"},
```

Reply:
Nope, I've given this my best effort and I can't do it. 

Feel free to check out my branch here: https://github.com/EliiseS/terraform-provider-azuredevops/tree/es/failed-test

I've gone ahead and removed the validation, since it wouldn't cover all the scopes either way, just the first one.

Reply:
Too bad 😞